libvast/src/concept/parseable/vast/expression.cpp

```diff
@@ -181,16 +181,23 @@ static auto make_expression_parser() {
     return dis.size() == 1 ? std::move(dis[0]) : expression{dis};
   };
   auto ws = ignore(*parsers::space);
-  auto negate_pred = [](predicate p) { return negation{expression{p}}; };
-  auto negate_expr = [](expression expr) { return negation{expr}; };
+  auto negate_expr = [](expression expr) { return negation{std::move(expr)}; };
   rule<Iterator, expression> expr;
   rule<Iterator, expression> group;
   // clang-format off
+  auto expand = [](data x) {
+    auto pred = to_value_predicate(std::move(x));
+    return caf::visit(expander{}, expression{std::move(pred)});
```

Comment:
Should we just call this inside of `to_value_predicate`? Why add another lambda on top?

Reply:
Or, the other way around: Should we call `to_value_predicate` inside the expander? Then its name makes sense.